.github/workflows/osx_build.yml

```diff
@@ -0,0 +1,52 @@
+name: MacOS build and test rolesanywhere-credential-helper
+on:
+  push:
+    branches: [ main ]
+  
+permissions:
+  statuses: write
+  id-token: write
+
+jobs:
+  OSX:
+    name: Build on ${{matrix.runner_image}}
+    runs-on: ${{matrix.runner_image}}
+    strategy:
+      matrix:
+        include:
+          - runner_image: macos-13
+            s3_path: X86_64/OSX/MacOS
+            display_name: "MacOS Intel"
+          - runner_image: macos-14
```

Comment:
What about backwards compatibility? Ideally, we would build on the oldest version that isn't EOL for each architecture, but do we have that level of control here?